types/index.d.ts

```diff
@@ -3620,34 +3623,19 @@ declare namespace math {
     randomSeed?: string | null;
   }
 
-  interface MathJsChain {
-    done(): any;
+  interface MathJsChain<TValue = any> {
+    done(): TValue;
 
     /*************************************************************************
      * Construction functions
      ************************************************************************/
 
-    /**
-     * Create a BigNumber, which can store numbers with arbitrary precision.
-     * When a matrix is provided, all elements will be converted to
-     * BigNumber.
-     */
-    bignumber(): MathJsChain;
-
-    /**
-     * Create a boolean or convert a string or number to a boolean. In case
-     * of a number, true is returned for non-zero numbers, and false in case
-     * of zero. Strings can be 'true' or 'false', or can contain a number.
-     * When value is a matrix, all elements will be converted to boolean.
-     */
-    boolean(): MathJsChain;
-
```

Comment:
Chainable methods are defined on the subtype(s) of `MathJsChain` that they are applicable to

Reply:
Chainable methods have their `this` property type overloaded with the type of `MathJSChain` that they are able to work on